### DISCUSSION TITLE
nested package rules

### DISCUSSION THREAD

**Discussion by sorrowfulwarrior** [renovatebot/renovate#20686] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.83.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

15.9.1

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I grouped dependencies into breaking and non-breaking updates. I use zero versioning dependencies and I want to set a rule in the way that patch updates are included in the non-breaking group and minor and major updates include in the breaking group. Is there any way to create a nested rule in package rules? 
desired result -> In the non-breaking group:
for >= 1.0.0 I like to get minor and patch update
for < 1.0.0 I like to get  just patch update
desired result -> In the breaking group:
for >= 1.0.0 I like to get major update
for < 1.0.0 I like to get minor and major update

Is there any solution to this?

### What you tried so far.

I tried to separate to use  `separateMinorPatch: true` but I could not reach the desired result.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Look into matchCurrentVersion

**Reply by sorrowfulwarrior** [reply]
Thank you for your response.
I already looked into it and came up with this idea but I think It does not affect.  
I can group them separately like `zero-version-non-breaking` and `non-breaking` but I want to have just one MR for non-breaking including minor and patch update for normal dependencies and patch update for zero-version dependencies.
```
groupName: "frontend",
      matchPaths: ["frontend/**"],
      rangeStrategy: "update-lockfile",
      labels: ["frontend", "In range"],
      automerge: false,
      enabled: true,
      minor: {
        matchCurrentVersion: "<1.0.0",
        enabled: false
      },
      major: {
        matchCurrentVersion: "<1.0.0",
        enabled: false
      },
```

**Reply by rarkins** [reply]
Don't nest minor and major objects within packageRules. matchCurrentVersion must appear directly within a package rule. Instead use matchUpdateTypes.

**Reply by sorrowfulwarrior** [reply]
Thank you I used this approach which I think would be helpful to share.
`enabled` can be configured with a CI Var so the desired group enables when the respected job is activated.
```
{
        matchPaths: ["backend/**"],
        matchCurrentVersion: "<1.0.0",
        rangeStrategy: "update-lockfile",
        patch: {
          groupName: "backend",
          labels: ["non-breaking"],
          enabled: true,
        },
        minor: {
          groupName: "backend",
          labels: ["non-breaking"],
          enabled: false,
          
        },
        major: {
          groupName: "backend",
          labels: ["breaking"],
          enabled: false,
        },
```
